Sources/ConfigCat/ConfigCatClient.swift

```diff
@@ -268,22 +271,24 @@ public final class ConfigCatClient: NSObject, ConfigCatClientProtocol {
                    Value.self != Any.self &&
                    Value.self != Any?.self {
             let message = "Only String, Integer, Double, Bool or Any types are supported."
-            log.error(message: message)
+            log.error(eventId: 2022, message: message)
             hooks.invokeOnFlagEvaluated(details: EvaluationDetails.fromError(key: key, value: defaultValue, error: message, user: evalUser))
             completion(TypedEvaluationDetails<Value>.fromError(key: key, value: defaultValue, error: message, user: evalUser))
             return
         }
         getSettings { result in
             if result.settings.isEmpty {
-                let message = String(format: "Config is not present. Returning defaultValue: [%@].", "\(defaultValue)")
-                self.log.error(message: message)
+                let message = String(format: "Config JSON is not present. Returning the `defaultValue` parameter that you specified in your application: '%@'.", "\(defaultValue)")
+                self.log.error(eventId: 1000, message: message)
                 self.hooks.invokeOnFlagEvaluated(details: EvaluationDetails.fromError(key: key, value: defaultValue, error: message, user: evalUser))
                 completion(TypedEvaluationDetails<Value>.fromError(key: key, value: defaultValue, error: message, user: evalUser))
                 return
             }
             guard let setting = result.settings[key] else {
-                let message = String(format: "Value not found for key '%@'. Here are the available keys: %@", key, [String](result.settings.keys))
-                self.log.error(message: message)
+                let message = String(format: "Failed to evaluate setting '%@' (the key was not found in config JSON). "
+                    + "Returning the `defaultValue` parameter that you specified in your application: '%@'. Available keys: %@.",
```

Comment:
Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`